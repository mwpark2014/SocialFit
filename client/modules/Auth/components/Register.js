import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { FormGroup, Input, Button,
  Label, Row, Col } from 'reactstrap';

export default class Register extends Component {
  constructor(props) {
    super(props);

    this.state = { name: '', username: '', email: '', password: '', password2: '' };
    this.handleSubmit = this.handleSubmit.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  handleInputChange(event) {
    const target = event.target;
    const value = target.value;
    const name = target.name;

    this.setState({
      [name]: value,
    });
  }

  handleSubmit(event) {
    event.preventDefault();
    this.props.registerUser({ username: this.state.username,
                   name: this.state.name,
                   email: this.state.email,
                   password: this.state.password });
    // Clear all the fields
    this.state = { name: '', username: '', email: '', password: '', password2: '' };
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <FormGroup>
          <Row>
            <Col md={4}>
              <Label>Name</Label>
            </Col>
            <Col md={7}>
              <Input
                type="text" name="name" placeholder="Name"
                onChange={this.handleInputChange} value={this.state.name} required
              />
            </Col>
          </Row>
        </FormGroup>
        <FormGroup>
          <Row>
            <Col md={4}>
              <Label>Username</Label>
            </Col>
            <Col md={7}>
              <Input
                type="text" name="username" placeholder="Username"
                onChange={this.handleInputChange} value={this.state.username} required
              />
            </Col>
          </Row>
        </FormGroup>
        <FormGroup>
          <Row>
            <Col md={4}>
              <Label>Email</Label>
            </Col>
            <Col md={7}>
              <Input
                type="email" name="email" placeholder="Email"
                onChange={this.handleInputChange} value={this.state.email} required
              />
            </Col>
          </Row>
        </FormGroup>
        <FormGroup>
          <Row>
            <Col md={4}>
              <Label>Password</Label>
            </Col>
            <Col md={7}>
              <Input
                type="password" name="password" placeholder="Password"
                onChange={this.handleInputChange} value={this.state.password} required
              />
            </Col>
          </Row>
        </FormGroup>
        <FormGroup>
          <Row>
            <Col md={4}>
              <Label>Confirm Password</Label>
            </Col>
            <Col md={7}>
              <Input
                type="password" name="password2" placeholder="Password"
                onChange={this.handleInputChange}
                value={this.state.password2} required
              />
            </Col>
          </Row>
        </FormGroup>
        <Row>
          <Col md={{ size: 7, offset: 4 }}>
            <Button type="submit" block>Register</Button>
          </Col>
        </Row>
      </form>
    );
  }
}

Register.propTypes = {
  errorMessage: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  registerUser: PropTypes.func.isRequired,
};
