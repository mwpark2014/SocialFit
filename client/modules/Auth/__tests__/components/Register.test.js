import React from 'react';
import Register from '../../components/Register';
import renderer from 'react-test-renderer';

const registerUser = jest.fn(); //eslint-disable-line

describe('Test render & styles', () => {
  test('Register component snapshot', () => {
    const component = renderer.create(
      <Register
        registerUser={registerUser} errorMessage=""
        message=""
      />,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot(); // eslint-disable-line
  });
});
