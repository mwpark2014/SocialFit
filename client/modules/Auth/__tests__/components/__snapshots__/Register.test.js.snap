// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render & styles Register component snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <div
      className="row"
    >
      <div
        className="col col-md-4"
      >
        <label
          className=""
          htmlFor={undefined}
        >
          Name
        </label>
      </div>
      <div
        className="col col-md-7"
      >
        <input
          className="form-control"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="row"
    >
      <div
        className="col col-md-4"
      >
        <label
          className=""
          htmlFor={undefined}
        >
          Username
        </label>
      </div>
      <div
        className="col col-md-7"
      >
        <input
          className="form-control"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          required={true}
          type="text"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="row"
    >
      <div
        className="col col-md-4"
      >
        <label
          className=""
          htmlFor={undefined}
        >
          Email
        </label>
      </div>
      <div
        className="col col-md-7"
      >
        <input
          className="form-control"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          required={true}
          type="email"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="row"
    >
      <div
        className="col col-md-4"
      >
        <label
          className=""
          htmlFor={undefined}
        >
          Password
        </label>
      </div>
      <div
        className="col col-md-7"
      >
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <div
      className="row"
    >
      <div
        className="col col-md-4"
      >
        <label
          className=""
          htmlFor={undefined}
        >
          Confirm Password
        </label>
      </div>
      <div
        className="col col-md-7"
      >
        <input
          className="form-control"
          name="password2"
          onChange={[Function]}
          placeholder="Password"
          required={true}
          type="password"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col col-md-7 offset-md-4"
    >
      <button
        className="btn btn-secondary btn-block"
        onClick={[Function]}
        type="submit"
      >
        Register
      </button>
    </div>
  </div>
</form>
`;
